src/pcre2_compile.c

```diff
@@ -2493,6 +2500,224 @@ return -1;
 
 
 
+/*************************************************
+*          Implement extended classes            *
+*************************************************/
+
+/* This function is called to test whether a series of META codes can be
+compiled to a simple class (OP_CLASS, OP_NCLASS, OP_XCLASS) or whether
+it requires an extended class (OP_ECLASS).
+
+Arguments:
+  ptr       points to the first META code, which should be one past the
+            META_CLASS or META_CLASS_NOT start code
+  pendptr   optional out-pointer to receive the end of the simple codes
+
+Returns:    TRUE if the pointed-to META_CLASS is simple, that is, we
+              reached a META_CLASS_END without encountering extended class
+              syntax. In this case, *pendptr is set to the position of the
+              META_CLASS_END itself.
+            FALSE otherwise. In this case, *pendptr is set to the position
+              of the META code that caused the function to return FALSE.
+
+In both cases, the span ptr...*pendptr is the longest sequence of simple
+codes starting at ptr.
+*/
+
+static BOOL
+check_simple_class(uint32_t *ptr, uint32_t **pendptr)
```

Comment:
Many of these class operations should go to `pcre2_compile_class.c`, it was created for this purpose.

Reply:
Great - are you pausing your work on character classes for a few days? I'll move the code into there, but it would be nice to wait until you're done with any major changes, or else do it quickly while things are quiet.

I can cope with some conflicts, just ideally not massive ones.

Reply:
Currently I have no plans to work on the compiler part. Maybe some jit optimizations.